src/fromager/commands/bootstrap.py

```diff
@@ -226,9 +228,14 @@ def write_constraints_file(
             output.write(
                 f"# ERROR: no single version of {dep_name} met all requirements\n"
             )
-            logging.error("%s: no single version meets all requirements", dep_name)
+            logger.error("%s: no single version meets all requirements", dep_name)
+            conflicting_deps.add(dep_name)
             for dv in sorted(versions):
                 output.write(f"{dep_name}=={dv}\n")
+    for dep_name in conflicting_deps:
+        logger.error("finding why %s was being used", dep_name)
+        for node in graph.get_nodes_by_name(dep_name):
+            find_why(graph, node, -1, 1, [])
```

Comment:
using `-1` here will cause `find_why` to print up till root level (i.e. for each dependency it will print parents, grandparents, great grandparents till the toplevel). If this is too much, I can change to only print the immediate parents.

Reply:
The `explain-duplicates` output gives all of the packages that depend on the duplicate, and the version ranges allowed in each case. Will this output give us that information clearly?

Reply:
`explain-duplicates` will only give the packages that directly depend on the duplicate. `find_why` gives this information and will also tell us where the package that depends on the duplicate comes from

Reply:
So we get the full dependency tree, including the actual requirements specifications at each step?

Reply:
yep including the req all the way upto the toplevel

Reply:
OK, sounds good